application/ui/components/input/array.jsx

```diff
@@ -0,0 +1,147 @@
+/** @jsx React.DOM */
+'use strict';
+
+var _             = require('underscore');
+var React         = require('react');
+
+module.exports = React.createClass({
+
+    displayName : 'ArrayParameter',
+
+    getInitialState : function()
+    {
+        return {
+            values   : [],
+            numItems : 0
```

Comment:
Is this field necessary? We could derive the number of items from `this.state.values.length`, and that way we avoid any potential bugs where `numItems` could be lying.